File: Day_25/Q098.cpp
#include<bits/stdc++.h>
using namespace std;

//Function To Find Common Characters
string common_chars(string str1,string str2){
    string ans = "";
    unordered_set<char> st;

    for(int i = 0;i < str1.length();i++){
        st.insert(str1[i]);
    }

    for(int i = 0;i < str2.length();i++){

        //Check Common Character
        if(st.find(str2[i]) != st.end()){
            ans += str2[i];
            st.erase(str2[i]);
        }
    }

    return ans;
}

int main(){
    string str1,str2;

    getline(cin,str1);
    getline(cin,str2);

    //Print Result
    cout << common_chars(str1,str2);

    return 0;
}

File: Day_25/Q098.java
import java.util.*;

class Q98{
    static Scanner sc = new Scanner(System.in);

    //Function To Find Common Characters
    static String common_chars(String str1,String str2){
        String ans = "";
        HashSet<Character> st = new HashSet<>();

        for(int i = 0;i < str1.length();i++){
            st.add(str1.charAt(i));
        }

        for(int i = 0;i < str2.length();i++){
            char ch = str2.charAt(i);

            //Check Common Character
            if(st.contains(ch)){
                ans += ch;
                st.remove(ch);
            }
        }

        return ans;
    }

    public static void main(String args[]){
        String str1 = sc.nextLine();
        String str2 = sc.nextLine();

        //Print Result
        System.out.println(common_chars(str1,str2));

        sc.close();
    }
}